### ISSUE TITLE
tree.DecisionTreeClassifier() model not splitting similarly to sklearn on glass data

### ISSUE THREAD

**Issue by frederikhoengaard** [frederikhoengaard/fhlearn#2]
some splits are different during fitting - suggest setting max depth to 3 and evaluate what is happening in the _find_best_split method

**Comment by frederikhoengaard** [comment]
Problem turned out to be caused by default random_state attribute in instance variables being set to 42. 

Now random_state is random by default and the model achieves similar scores as sklearn. With random state at 36 the model has 80% accuracy on the glass test data with no hyperparameters.

**Comment by frederikhoengaard** [comment]
Turns out sklearn handles float values a bit unexpectedly sometimes. This leads to actual threshold values being slightly different than what one might expect and what they might appear as in an exported graph due to rounding. This is probably not so easy to address as it completely depends on the data.
